pkg/resource/interfaces.go

```diff
@@ -210,3 +211,29 @@ type Workload interface {
 
 	Conditioned
 }
+
+// Composite resource manages one or more Composable resources.
+type Composite interface {
+	Object
+
+	Conditioned
+	Bindable
+	ConnectionSecretWriterTo
+
+	SetCompositionSelector(*v1.LabelSelector)
+	GetCompositionSelector() *v1.LabelSelector
+
+	SetCompositionReference(*corev1.ObjectReference)
+	GetCompositionReference() *corev1.ObjectReference
+
+	SetResourceReferences([]corev1.ObjectReference)
+	GetResourceReferences() []corev1.ObjectReference
```

Comment:
Should we follow the existing pattern of declaring these last three groups of getters and setters as smaller interfaces and composing them into the larger one? Maybe `CompositionSelector`, `CompositionReferencer`, and `ComposedResourcesReferencer`?